docs/CONTRIBUTING_DOCS.md

```diff
@@ -38,226 +36,226 @@ pip install -r docs/requirements.txt
+to include code blocks. 
 
-```
-.. code-block:: python
+### Just the code, don't evaluate
+If you want to include a code snippet that doesn't get run (but has syntax
+highlighting), use the `code-block` directive:
 
-   1+1        # simple example
+````
+```{code-block} python
 
+   1+1        # simple example
 ```
-
+````
+### Run the code with doctest
 In order to make sure that the block is parsed with `make doctest`, use the
 `testcode` directive. This can be used in pair with `testoutput`, if something
 is printed, and, eventually `testsetup`, to import modules or set up variables
 in an invisible block. An example is:
 
-```
-.. testcode:: python
+````
+```{testcode} python
 
    1+1        # simple example
-
 ```
+````
 with no output and
 
-```
-.. testcode:: python
-
+````
+```{testcode} python
    print(1+1)        # explicitly print
+```
 
-.. testoutput:: python
-
+```{testoutput} python
    2        # match the print message
-
-
 ```
+````
 
-The use of `testsetup` allows blocks that do not render:
-
-```
-.. testsetup:: python
+If you have code blocks you want to run, but not be displayed, use the 
+`testsetup` directive:
 
+````
+```{testsetup} python
    import numpy as np  # this block is not rendered in the html or pdf
+```
 
-.. testcode:: python
-
+```{testcode} python
    np.array(2)
+```
 
-.. testoutput:: python
-
+```{testoutput} python
    array(2)
-
 ```
-
+````
+### IPython code blocks
 There is also the `doctest` directive, which allows to include interactive
 Python blocks. These need to be given this way:
 
-```
-.. doctest:: python
-
+````md
+```{doctest} python
    >>> import numpy as np
    >>> print(np.array(2))
    array(2)
-
-Notice that no space is left between the last input and the output.
-```
- A way to test docstrings without installing sphinx is with [`pytest` +
- `doctest`](http://doc.pytest.org/en/latest/doctest.html):
-
-```bash
-pytest --doctest-glob='*.rst'
 ```
-or alternatively
-
-```bash
-pytest --doctest-modules
+````
+```{note}
+Notice that no space is left between the last input and the output when writing code blocks with interactive inputs and outputs.
 ```
 
-However, this only checks `doctest` blocks, and does not recognize `testcode`
-blocks. Moreover, it does not parse the `conf.py` file nor uses sphinx.
-A way to include testing of `testcode` and `testoutput` blocks is with the
-[`pytest-sphinx`](https://github.com/thisch/pytest-sphinx) plugin. Once
-installed,
-```bash
-pip install pytest-sphinx
-```
-it will show up as a plugin, just like `pytest-coverage` and others, simply
-calling
-```bash
-pytest --doctest-glob='*.rst'
-```
-The `pytest-sphinx` plugin does not support `testsetup` directives.
+### Skipping or ignoring a test
 
 In order to skip a test, if this is problematic, one can use the `SKIP` and
 `IGNORE` keywords, adding them as comments next to the relevant line or block:
 
 ```
 >>> something_that_raises()  # doctest: +IGNORE
```

Comment:
```suggestion
>>> something_that_raises()  #: doctest: +IGNORE
```
```suggestion
>>> something_that_raises()  # doctest: +IGNORE
```

Reply:
To show the comment in the docs, `#: doctest: +IGNORE`

Reply:
Good catch!